DependencyInjection/M6WebKafkaExtension.php

```diff
@@ -0,0 +1,206 @@
+     */
+    public function getRdKafkaConf(): \RdKafka\Conf
+    {
+        return new \RdKafka\Conf();
+    }
+
+    /**
+     * @return \RdKafka\TopicConf
+     */
+    public function getTopicConf(): \RdKafka\TopicConf
+    {
+        return new \RdKafka\TopicConf();
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param array            $config
+     */
+    protected function loadProducers(ContainerBuilder $container, array $config)
+    {
+        foreach ($config['producers'] as $key => $producer) {
+            $producerDefinition = $this->getDefinition('M6Web\Bundle\KafkaBundle\RdKafkaProducerManager');
+
+            $this->setEventDispatcher($config, $producerDefinition);
+
+            $conf = $this->getReadyRdKafkaConf($producer['conf']);
+            $rdKafkaProducer = new \RdKafka\Producer($conf);
+
+            $producerDefinition->addMethodCall('setRdKafkaProducer', [$rdKafkaProducer]);
+            $this->setProducerManager($producer, $producerDefinition);
+
+            $container->setDefinition(
+                sprintf('m6_web_kafka.producer.%s', $key),
+                $producerDefinition
+            );
+        }
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param array $config
+     */
+    protected function loadConsumers(ContainerBuilder $container, array $config)
+    {
+        foreach ($config['consumers'] as $key => $consumer) {
+            $consumerDefinition = $this->getDefinition('M6Web\Bundle\KafkaBundle\RdKafkaConsumerManager');
+
+            $this->setEventDispatcher($config, $consumerDefinition);
+
+            $topicConf = $this->getReadyTopicConf($consumer['topicConf']);
+            $conf = $this->getReadyRdKafkaConf($consumer['conf']);
+            $conf->setDefaultTopicConf($topicConf);
+
+            $kafkaConsumer = new \RdKafka\KafkaConsumer($conf);
+
+            $consumerDefinition->addMethodCall('setRdKafkaKafkaConsumer', [$kafkaConsumer]);
+            $consumerDefinition->addMethodCall('addTopic', [$consumer['topics'], $kafkaConsumer]);
+            $consumerDefinition->addMethodCall('setTimeoutConsumingQueue', [$consumer['timeout_consuming_queue']]);
+
+            $container->setDefinition(
+                sprintf('m6_web_kafka.consumer.%s', $key),
+                $consumerDefinition
+            );
+        }
+    }
+
+    /**
+     * @param array $config
+     * @param $entityDefinition
+     */
+    protected function setEventDispatcher(array $config, $entityDefinition)
+    {
+        if ($config['event_dispatcher'] === true) {
+            $entityDefinition->addMethodCall(
+                'setEventDispatcher',
+                [
+                    new Reference('event_dispatcher'),
+                ]
+            );
+        }
+    }
+
+    /**
+     * @param array      $entity
+     * @param Definition $entityDefinition
+     */
+    protected function setProducerManager(array $entity, Definition $entityDefinition)
```

Comment:
configProducerManager ? or why not getProducerManager with the getDefinition inside and all the addMethodCall ?